### PR TITLE
Issue#4

### PR DESCRIPTION
**Pull request by gurjar-mehul**
@reneweteling  - Please review. 

Mehul

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -29,6 +29,8 @@ group :development do
   gem 'quiet_assets'
   gem 'better_errors'
   gem 'binding_of_caller'
+  # Help to add/remove n+1 query.
+  gem 'bullet'
```
Comment (reneweteling) [inline Gemfile:32]: gems are self explanatory, so dont add comments for things easily google'd or common knowledge

File: app/admin/beer.rb [hunk 2]
```diff
@@ -25,4 +25,11 @@
     column :created_at
   end
 
+  # Adding N + 1 for Beer index method.
+  controller do
```
Comment (reneweteling) [inline app/admin/beer.rb:28]: same

File: app/admin/user.rb [hunk 3]
```diff
@@ -21,6 +21,16 @@ def update
       end
       super
     end
+
+    def destroy
+      # Before destroy, Checking associated
+      user = User.find(params[:id])
```
Comment (reneweteling) [inline app/admin/user.rb]: isnt this default behaviour? I always get errors without exending this function. Could you find out what the differance is between this function and just the default behaviour

File: app/models/user.rb [hunk 4]
```diff
@@ -1,9 +1,9 @@
 class User < ActiveRecord::Base
   devise :database_authenticatable, :validatable, :token_authenticatable
 
-  has_many :beers, dependent: :destroy
-  has_many :transactions, dependent: :destroy
-  
+  # Before destroy, Checking associated
+  has_many :beers, dependent: :restrict_with_error
```
Comment (reneweteling) [inline app/models/user.rb:4]: to many comments :)

File: config/environments/development.rb [hunk 5]
```diff
@@ -38,4 +38,14 @@
...
+  # Bullet
+  config.after_initialize do
+    Bullet.enable = true
+    Bullet.bullet_logger = true
+    Bullet.console = true
+  #  Bullet.growl = true
+    Bullet.rails_logger = true
+    Bullet.add_footer = true
+  end
 end
```
Comment (reneweteling) [inline config/environments/development.rb:51]: 👍

**Comment by reneweteling** [comment]
@05BIT008 check out my comments, and fix the test suite. than you can merge it

File: Gemfile (hunk 1 above)
Comment (gurjar-mehul) [inline Gemfile:32]: Removed.

File: app/admin/beer.rb (hunk 2 above)
Comment (gurjar-mehul) [inline app/admin/beer.rb:28]: Removed.

File: app/admin/user.rb (hunk 3 above)
Comment (gurjar-mehul) [inline app/admin/user.rb]: I thought to add custom message. Yes default behaviour work well. 
Removed unused code and use default behaviour.

File: app/models/user.rb (hunk 4 above)
Comment (gurjar-mehul) [inline app/models/user.rb:4]: Removed :)

**Pull request merged**